client/src/components/menubar/authButtons.js

```diff
@@ -1,31 +1,122 @@
-import React from "react";
-import { AnchorButton, ButtonGroup, Tooltip } from "@blueprintjs/core";
+import React, { useState } from "react";
+import {
+  AnchorButton,
+  ButtonGroup,
+  Elevation,
+  Tooltip,
+  Popover,
+  PopoverPosition,
+  Checkbox,
+  Card,
+  Button,
+  Classes,
+} from "@blueprintjs/core";
 import * as globals from "../../globals";
 import styles from "./menubar.css";
 
+import { storageGet, storageSet, KEYS } from "../util/localStorage";
+
+const LOGIN_PROMPT_OFF = "off";
+
 const Auth = React.memo((props) => {
+  const [isPromptOpen, setIsPromptOpen] = useState(shouldShowPrompt());
+
   const { auth, userinfo } = props;
 
-  if (!auth || (auth && !auth.requires_client_login)) return null;
+  const isAuthenticated = userinfo && userinfo.is_authenticated;
+
+  if (shouldShowAuth()) return null;
+
+  const AuthButton = () => {
+    return (
+      <AnchorButton
+        type="button"
+        data-testid="auth-button"
+        disabled={false}
```

Comment:
nit: I think we could just omit this if it's false

Reply:
Good point! Done 😆